hax/hax/motr/delivery.py

```diff
@@ -75,6 +84,7 @@ def wait_for_any(self,
         """
         condition = Condition()
         with self.lock:
+            self.groom_unsorted(promise)
```

Comment:
Okay, we are grooming the ones on which the sender does not wait on, but what about already delivered? we are still waiting for delivery but a notification will never come right? don't we need to check in recently_delivered before waiting on the condition post grooming?

Reply:
Not sure that I understood you right.

`wait_for_*` functions block on Condition with a timeout. If the delivery will never happen, the waiting thread is unblocked by the timeout, and then [`_verify_delivered()`](https://github.com/Seagate/cortx-hare/blob/2e0ae5a413b6e1109102f900fea132ec274cf1c1/hax/hax/motr/delivery.py#L54) is invoked. Within that function `self.waiting_clients[promise]` is removed, so we don't have a leak here.

Do I miss anything?

Reply:
And one more thing. Please note that `groom_unsorted()` moves the entries that relate to this particular promise to `recently_delivered`:
https://github.com/Seagate/cortx-hare/blob/2e0ae5a413b6e1109102f900fea132ec274cf1c1/hax/hax/motr/delivery.py#L129-L138

Reply:
I mean for the cases where the message is already delivered before a sender invokes wait_for_any or wait_for_all, in such a case we are not checking if the message is already delivered, don't we need to check the `unsorted_deliveries` before going into wait?

Reply:
Good point. I'll need to fix that.

Reply:
Fixed in #1627. Not relevant for this PR anymore.